src/formatDate.js

```diff
@@ -50,7 +50,46 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
+  const newDate = [];
+  const year = [];
+
+  for (let i = 0; i < 3; i++) {
+    if (fromFormat[i].includes('Y')) {
+      year.push(dateArray[i]);
+      year.push(fromFormat[i]);
+      break;
+    }
+  }
+
+  year.push(toFormat.find(a => a.includes('Y')));
+
+  if (year[1].length === 4 && year[2].length === 2) {
+    year[0] = year[0].slice(2);
+  }
+
+  if (year[1].length === 2 && year[2].length === 4) {
+    const yearShort = year[0];
+
+    if (yearShort < 30) {
+      year[0] = '20' + yearShort;
+    } else {
+      year[0] = '19' + yearShort;
+    }
+  }
```

Comment:
Move this year format converting logic to separate function

Reply:
Also would be nice to simplify it, it looks too complicated now. I think you can do it even without array creating